validation-test/stdlib/CollectionDiagnostics.swift

```diff
@@ -58,11 +58,11 @@ func sortResultIgnored<
   sequence.sorted() // expected-warning {{result of call to 'sorted()' is unused}}
   sequence.sorted { $0 < $1 } // expected-warning {{result of call to 'sorted(isOrderedBefore:)' is unused}}
 
-  mutableCollection.sorted() // expected-warning {{result of call to non-mutating function 'sorted()' is unused; use 'sort()' to mutate in-place}} {{21-27=sort}}
-  mutableCollection.sorted { $0 < $1 } // expected-warning {{result of call to non-mutating function 'sorted(isOrderedBefore:)' is unused; use 'sort(isOrderedBefore:)' to mutate in-place}} {{21-27=sort}}
+  mutableCollection.sorted() // expected-warning {{result of call to 'sorted()' is unused}}
+  mutableCollection.sorted { $0 < $1 } // expected-warning {{result of call to 'sorted(isOrderedBefore:)' is unused}}
 
-  array.sorted() // expected-warning {{result of call to non-mutating function 'sorted()' is unused; use 'sort()' to mutate in-place}} {{9-15=sort}}
-  array.sorted { $0 < $1 } // expected-warning {{result of call to non-mutating function 'sorted(isOrderedBefore:)' is unused; use 'sort(isOrderedBefore:)' to mutate in-place}} {{9-15=sort}}
+  array.sorted() // expected-warning {{result of call to 'sorted()' is unused}}
+  array.sorted { $0 < $1 } // expected-warning {{result of call to 'sorted(isOrderedBefore:)' is unused}}
```

Comment:
Yes. That was noted in the proposal (SE-0047) as a downside. To partially make up for this, there are now `mutatingVariant` and `nonmutatingVariant` doc tags that interact with code completion. I added those in #2204.